### PR TITLE
This fixes problem with tagging image for external registry

### PR DESCRIPTION
**Pull request by phracek**
See error:
https://gist.github.com/rhscl-automation/15901ed849d46c176c7b13afa64e8abf

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Comment by phracek** [comment]
Code snip:

```bash
VERSIONS="10 12 14" SKIP_SQUASH=0 UPDATE_BASE=1 OS=rhel7 CLEAN_AFTER= DOCKER_BUILD_CONTEXT=. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" TEST_OPENSHIFT_4=true /usr/bin/env bash common/tag.sh
-> Tagging image 'b2b3f9da90ca81195e1d02c14d9edfc7893f4dc90d64867ff73147b3b584dde1' as 'rhscl/nodejs-10-rhel7:1' and 'rhscl/nodejs-10-rhel7:latest' and 'rhscl/nodejs-10-rhel7:20200915-51a50b7'
-> Tagging image 'fd7a702ee9ea9482e1d5159d00d3be3bc78235ba2ee66937322b88b516bbe374' as 'rhscl/nodejs-12-rhel7:1' and 'rhscl/nodejs-12-rhel7:latest' and 'rhscl/nodejs-12-rhel7:20200915-51a50b7'
-> Image for version 14 not built, skipping tag.
VERSIONS="10 12 14" BASE_IMAGE_NAME="nodejs" SKIP_SQUASH=0 UPDATE_BASE=1 OS=rhel7 CLEAN_AFTER= DOCKER_BUILD_CONTEXT=. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" TEST_OPENSHIFT_4=true /usr/bin/env bash common/test-remote-cluster.sh
Setting OCP4 client
[snip]
Now using project "sclorg-test-21428" on server "https://api.core-serv-ocp.hosted.psi.rdu2.redhat.com:6443".
You can add applications to this project with the 'new-app' command. For example, try:
    oc new-app ruby~https://github.com/sclorg/ruby-ex.git
to build a new example application in Python. Or use kubectl to deploy a simple Kubernetes application:
    kubectl create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node
secret/6340056-core-services-apps-ci-pull-secret created
Uploading image registry.access.redhat.com/rhscl/nodejs-10-rhel7 as docker-registry.upshift.redhat.com/rhscl-ci-testing/nodejs-10-rhel7 , nodejs-10-rhel7 into external registry.
loging
Login Succeeded
REPOSITORY                                        TAG                 IMAGE ID            CREATED             SIZE
rhscl/nodejs-12-rhel7                             1                   fd7a702ee9ea        39 seconds ago      560 MB
rhscl/nodejs-12-rhel7                             20200915-51a50b7    fd7a702ee9ea        39 seconds ago      560 MB
rhscl/nodejs-12-rhel7                             latest              fd7a702ee9ea        39 seconds ago      560 MB
rhscl/nodejs-12-rhel7                             squashed            fd7a702ee9ea        39 seconds ago      560 MB
rhscl/nodejs-12-rhel7                             raw                 8fde941aaf76        39 seconds ago      562 MB
rhscl/nodejs-10-rhel7                             1                   b2b3f9da90ca        2 minutes ago       507 MB
rhscl/nodejs-10-rhel7                             20200915-51a50b7    b2b3f9da90ca        2 minutes ago       507 MB
rhscl/nodejs-10-rhel7                             latest              b2b3f9da90ca        2 minutes ago       507 MB
rhscl/nodejs-10-rhel7                             squashed            b2b3f9da90ca        2 minutes ago       507 MB
rhscl/nodejs-10-rhel7                             raw                 989d6b21c9e2        2 minutes ago       508 MB
registry.access.redhat.com/rhscl/s2i-base-rhel7   1                   414d8f0fdc34        7 weeks ago         472 MB
Error response from daemon: no such id: registry.access.redhat.com/rhscl/nodejs-10-rhel7
OpenShift tests for registry.access.redhat.com/rhscl/nodejs-10-rhel7 failed.
```

File: test-lib-remote-openshift.sh [hunk 1]
```diff
@@ -65,13 +65,18 @@ function ct_os_upload_image_external_registry() {
...
+  local image_name_without_register
 
   ct_os_login_external_registry
 
   output_name="${INTERNAL_DOCKER_REGISTRY}/rhscl-ci-testing/$imagestream"
 
+  # This command removes 'registry.access.redhat.com' from 'registry.access.redhat.com/rhscl/nodejs-10-rhel7'
+  # image_name_without_register is now 'rhscl/nodejs-10-rhel7'
+  # Images from build and tag section have the name 'rhscl/nodejs-10-rhel7'
+  image_name_without_register=$(cut -d '/' -f 2- <<< "$input_name")
```
Comment (pkubatrh) [inline test-lib-remote-openshift.sh:77]: How about `${input_name#*/}` instead? Seems a bit more readable to me and would be done the same way as setting `image_name`.

Comment (phracek) [reply test-lib-remote-openshift.sh]: What about if `input_name` is `rhscl/nodejs-10-rhel7`? Then the input name will be `nodejs-10-rhel7`.

Comment (phracek) [reply test-lib-remote-openshift.sh]: I have updated to `grep` command. If `input_name` is `rhscl/nodejs-10-rhel7` or `registry.access.redhat.com/rhscl/nodejs-10-rhel7` then `image_name_without_register` is `rhscl/nodejs-10-rhel7`.

**Comment by phracek** [comment]
@pkubatrh PTAL. I have tested [test-openshift] and [test-openshit-4] on a resalloc machine and the change works properly.

**Pull request merged**